evm/vm/forks/sharding/shard_tracker.py

```diff
@@ -0,0 +1,157 @@
+import copy
+
+from cytoolz import (
+    pipe,
+)
+
+from eth_utils import (
+    event_signature_to_log_topic,
+    to_dict,
+    to_tuple,
+)
+
+from evm.rlp.headers import (
+    CollationHeader,
+)
+
+from evm.utils.hexadecimal import (
+    encode_hex,
+    decode_hex,
+)
+from evm.utils.numeric import (
+    big_endian_to_int,
+)
+
+
+class NextLogUnavailable(Exception):
+    pass
+
+
+class NoCandidateHead(Exception):
+    pass
+
+
+@to_dict
+def parse_collation_added_log(log):
+    # here assume `shard_id` is the first indexed , which is the second element in topics
+    shard_id_bytes32 = log['topics'][1]
+    data_hex = log['data']
+    data_bytes = decode_hex(data_hex)
+    score = big_endian_to_int(data_bytes[-32:])
+    is_new_head = bool(big_endian_to_int(data_bytes[-64:-32]))
+    header_bytes = shard_id_bytes32 + data_bytes[:-64]
+    collation_header = CollationHeader.from_bytes(header_bytes)
+    yield 'header', collation_header
+    yield 'is_new_head', is_new_head
+    yield 'score', score
```

Comment:
Since this function has no conditional keys, I think the use of `to_dict` reduces readability.  Maybe switch to just building the dictionary locally and returing it.

Reply:
No problem. I have changed it. Thank you